### PR TITLE
Hw26.1

### PR DESCRIPTION
**Pull request by margarita-ignatieva**

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.internet.shop.library.Injector;
+import com.internet.shop.service.interfaces.UserService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class UserController extends HttpServlet {
```
Comment (KovalAlexey) [inline src/main/java/com/internet/shop/controllers/UserController.java]:
```suggestion
public class GetAllUsersController extends HttpServlet {
```

File: src/main/webapp/WEB-INF/views/product/all.jsp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                <c:out value="${product.price}"/>
+            </td>
+            <td>
+                <a href = "${pageContext.request.contextPath}/products/add?id=${product.id}"> Buy </a>
+            </td>
+        </tr>
+    </c:forEach>
+</table>
+</body>
+</html>
```
Comment (KovalAlexey) [inline src/main/webapp/WEB-INF/views/product/all.jsp]: add empty line

File: src/main/webapp/WEB-INF/views/shoppingCart/allProducts.jsp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        </c:forEach>
+</table>
+<br>
+<button type = "submit"> Complete order </button>
+</form>
+<br>
+<br>
+<a href = "${pageContext.request.contextPath}/main"> Main Menu </a>
+</body>
+</html>
```
Comment (KovalAlexey) [inline src/main/webapp/WEB-INF/views/shoppingCart/allProducts.jsp]: add empty line

File: src/main/webapp/web.xml [hunk 4]
```diff
@@ -4,12 +4,74 @@
...
+        <servlet-name>buyProduct</servlet-name>
+        <url-pattern>/product/buy</url-pattern>
+    </servlet-mapping>
+    <servlet>
+        <servlet-name>currentShoppingCart</servlet-name>
+        <servlet-class>com.internet.shop.controllers.CartGetAllProductsController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>currentShoppingCart</servlet-name>
+        <url-pattern>/cart</url-pattern>
```
Comment (KovalAlexey) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/shopping-cart/products</url-pattern>
```

File: src/main/webapp/web.xml [hunk 5]
```diff
@@ -4,12 +4,74 @@
...
+        <servlet-name>allProducts</servlet-name>
+        <url-pattern>/product/all</url-pattern>
+    </servlet-mapping>
+    <servlet>
+        <servlet-name>buyProduct</servlet-name>
+        <servlet-class>com.internet.shop.controllers.CartProductAddingController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>buyProduct</servlet-name>
+        <url-pattern>/product/buy</url-pattern>
```
Comment (KovalAlexey) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/shopping-cart/products/add</url-pattern>
```

File: src/main/webapp/web.xml [hunk 6]
```diff
@@ -4,12 +4,74 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>registration</servlet-name>
+        <servlet-class>com.internet.shop.controllers.RegistrationController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>registration</servlet-name>
+        <url-pattern>/registration</url-pattern>
```
Comment (KovalAlexey) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/user/registration</url-pattern>
```

File: src/main/webapp/WEB-INF/views/product/all.jsp (hunk 2 above)
Comment (margarita-ignatieva) [reply src/main/webapp/WEB-INF/views/product/all.jsp]: thanks, missed that

File: src/main/java/com/internet/shop/controllers/RegistrationController.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import com.internet.shop.service.interfaces.UserService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class RegistrationController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private static UserService userService =
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/RegistrationController.java]: why service is static? 
```suggestion
    private UserService userService =
```

File: src/main/java/com/internet/shop/controllers/ProductGettingAllController.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.interfaces.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ProductGettingAllController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ProductGettingAllController.java]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/ProductAddingController.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+import com.internet.shop.library.Injector;
+import com.internet.shop.model.Product;
+import com.internet.shop.service.interfaces.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ProductAddingController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ProductAddingController.java]:
```suggestion
public class AddProductController extends HttpServlet {
```
let's change other controllers in the same style

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**